core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1593,6 +1594,21 @@ public void addView(View v) throws IOException {
         viewGroupMixIn.addView(v);
     }
 
+    /**
+     * Completely replaces views
+     */
+    public void setViews(Collection<View> views) throws IOException {
```

Comment:
As you noted somewhere else this is so that I can set the exact list of views from system-config-dsl plugin.

It's just that the mechanism I handle `Jenkins` is different from how I handle `Describable`s. The former already exists and I fiddle with its setters, the latter I use `DescribableModel` in the structs plugin.